pkg/operator/reports_test.go

```diff
@@ -335,3 +335,97 @@ func TestValidateReport(t *testing.T) {
+			expectErr: true,
+		},
+		{
+			name:      "valid report with an unset spec.Schedule field returns nil",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectErr: false,
+		},
+		{
+			name:      "invalid schedule with a set spec.Schedule field returns error",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, invalidSchedule, false),
+			expectErr: true,
+		},
+		{
+			name:      "valid schedule with a set spec.Schedule field and an unset Spec.Status.LastReportTime returns nil",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, validSchedule, false),
+			expectErr: false,
+		},
+		{
+			name:      "valid schedule with a set spec.Schedule field and a set Spec.Status.LastReportTime returns nil",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{LastReportTime: lastReportTime}, validSchedule, false),
+			expectErr: false,
+		},
+		{
+			name:      "valid schedule with a set spec.Schedule field and an unset Spec.Status.LastReportTime and a set Spec.ReportingStart returns nil",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, validSchedule, false),
+			expectErr: false,
+		},
+		{
+			name:      "valid schedule with a set spec.Schedule field and an unset Spec.Status.LastReportTime and an unset Spec.ReportingStart returns nil",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, nil, reportEnd, v1alpha1.ReportStatus{}, validSchedule, false),
+			expectErr: false,
+		},
+		{
+			name:      "valid schedule with a set spec.Schedule field and an unset Spec.Status.LastReportTime and a set Spec.NextReportTime returns nil",
+			report:    testhelpers.NewReport(testReportName, testNamespace, testQueryName, nil, reportEnd, v1alpha1.ReportStatus{NextReportTime: nextReportTime}, validSchedule, false),
+			expectErr: false,
+		},
+		{
+			name:        "unset Spec.Schedule with reportPeriod.periodStart > reportPeriod.periodEnd returns panic",
+			report:      testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportEnd, reportStart, v1alpha1.ReportStatus{NextReportTime: nextReportTime}, nil, false),
+			expectErr:   false,
+			expectPanic: true,
+		},
+	}
+
+	for _, testCase := range testTable {
+		var mockLogger = logrus.New()
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			if testCase.expectPanic {
+				assert.Panics(t, func() { getReportPeriod(time.Now(), mockLogger, testCase.report) }, "expected the test case would panic, but it did not")
+			} else {
+				_, err := getReportPeriod(time.Now(), mockLogger, testCase.report)
+				if testCase.expectErr {
```

Comment:
Need an else branch to cover when expectErr is false (probably just `assert.IsNil`